### PR TITLE
new eigenvalue solver: PRIMME

### PR DESCRIPTION
**Pull request by vlukes**
PRIMME: PReconditioned Iterative MultiMethod Eigensolver

https://github.com/primme/primme
https://www.cs.wm.edu/~andreas/software/

Easy installation: `pip install primme`

This solver gives almost the same results as `eig.matlab`:

Example #982:

========================== PRIMME ==========================
sfepy: number |         eigenvalue |  angular frequency |          frequency
sfepy:      1 | 5.485692258652e+03 | 7.406545928199e+01 | 1.178788395710e+01
sfepy:      2 | 5.532072378601e+03 | 7.437790248858e+01 | 1.183761083787e+01
sfepy:      3 | 5.322842318857e+04 | 2.307128587413e+02 | 3.671909190354e+01
sfepy:      4 | 2.815210206038e+05 | 5.305855450385e+02 | 8.444531222598e+01
sfepy:      5 | 2.830045827988e+05 | 5.319817504378e+02 | 8.466752521685e+01
sfepy:      6 | 3.832368111708e+05 | 6.190612337813e+02 | 9.852665543286e+01
sfepy:      7 | 9.660630368982e+05 | 9.828850578263e+02 | 1.564310154442e+02
sfepy:      8 | 9.748235008586e+05 | 9.873315050471e+02 | 1.571386894986e+02

========================== MATLAB ==========================
sfepy: number |         eigenvalue |  angular frequency |          frequency
sfepy:      1 | 5.485692258652e+03 | 7.406545928199e+01 | 1.178788395710e+01
sfepy:      2 | 5.532072378601e+03 | 7.437790248858e+01 | 1.183761083787e+01
sfepy:      3 | 5.322842318857e+04 | 2.307128587413e+02 | 3.671909190354e+01
sfepy:      4 | 2.815210206038e+05 | 5.305855450385e+02 | 8.444531222598e+01
sfepy:      5 | 2.830045827988e+05 | 5.319817504378e+02 | 8.466752521685e+01
sfepy:      6 | 3.832368111708e+05 | 6.190612337813e+02 | 9.852665543286e+01
sfepy:      7 | 9.660630368982e+05 | 9.828850578263e+02 | 1.564310154442e+02
sfepy:      8 | 9.748235302262e+05 | 9.873315199193e+02 | 1.571386918656e+02

### CODE REVIEW

**Comment by rc** [comment]
Great ,thanks! 

Can you rebase to fix the conflicts? `setup.py` was changed in #960.

File: setup.py [hunk 1]
```diff
@@ -88,6 +88,13 @@ def _scikit_umfpack_version(pkg_name):
     except:
         return None
 
+
+def _primme_version(pkg_name):
+    import primme
+
+    return '3.0.0'
```
Comment (rc) [inline setup.py:95]: OK. Most versions in `sfepy/version.py` are too old anyway, so maybe it is time to revisit or ditch those checks?

**Comment by rc** [comment]
I am playing with `sfepy/tests/test_eigenvalue_solvers.py`:

- It seems that the default `tol` cannot be `None` - it should be 0. 
- `eigenvectors` argument of `PrimmeEigenvalueSolver.__call__()` should be passed to the solver.
- Maybe use `conf.which.upper()` to allow also `'sm'` etc. for convenience?
- Can you add it there? The following works for me:
```
    'evp6' : ('eig.primme', {
        'maxiter' : 200,
        'tol' : 1e-10,
        'which' : 'SA',
    }),
<snip>
can_fail = ['eig.slepc', 'eig.matlab', 'eig.primme']
```

BTW. it fails when using 'SM' - the double eigenvalue is missed and computed as a single one. 'SA' works.

**Comment by rc** [comment]
Thanks! I can add the solver to the tests if you want?

**Comment by vlukes** [comment]
Yes, please do it.

**Pull request merged**